### PR TITLE
fix empty response from NLU

### PR DESCRIPTION
**Pull request by rameshpoomalai**
updated the code to display error message when there is no response from NLU/Error

### CODE REVIEW

File: src/main/java/com/ibm/watson/nlu/DemoServlet.java [hunk 1]
```diff
@@ -89,65 +87,31 @@ protected void doPost(final HttpServletRequest req, final HttpServletResponse re
...
+
 			//resp.getWriter().println("{ \"language\": \"en\", \"entities\": [ { \"type\": \"Merchant\", \"text\": \"DUNKI DONUTS\", \"count\": 1 }, { \"type\": \"Location\", \"text\": \"Girgaum\", \"count\": 1 },"+
 			//" { \"type\": \"Offer\", \"text\": \"Get 3 FREE\", \"count\": 1 }, { \"type\": \"Offer_Period\", \"text\": \"Valid till 15 Feb 2017\", \"count\": 1 }, { \"type\": \"Term_and_Conditions\", \"text\": \"T&C\", \"count\": 1 } ] }");
 
 		} catch (Exception e) {
-		    // Don't throw the error. Stuff it in an "Error" entity so the user can see it.
 			logger.log(Level.SEVERE, "Service error: " + e.getMessage(), e);
-			resp.getWriter().println("{ \"language\": \"en\", \"entities\": [ { \"type\": \"Error\", \"text\": \"" +
-					e.toString() + "\", \"count\": 1 } ] }");
+            resp.sendError(HttpStatus.SC_BAD_GATEWAY, e.getMessage() );
```
Comment (stevemar) [inline src/main/java/com/ibm/watson/nlu/DemoServlet.java]: the indenting here seems off to me

Comment (rameshpoomalai) [reply src/main/java/com/ibm/watson/nlu/DemoServlet.java]: Sorry,Let me fix the indent issue

**Review by rhagarty — CHANGES_REQUESTED** [review]
This change breaks running locally.
'getConfigParams()' is required to pull the user entered creds and modelID from config.properties.

**Comment by rameshpoomalai** [comment]
Hi Rich, I have. merged the content which is lost during last checkin and fixed the error handling for short text. Can you please review it?

**Comment by rhagarty** [comment]
Hello @rameshpoomalai - tested this locally and I don't like the error messages. First off, always displaying "Error 502" doesn't seem very helpful. Secondly, the error for bad model ID is no obvious what the problem is.

Here is a comparison of the errors shown for bad credentials and model ID not found:

Previous Version -
Error:com.ibm.watson.developer_cloud.service.exception.UnauthorizedException: Unauthorized: Access is denied due to invalid credentials
Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found

Your proposed version - 
Error 502: Unauthorized: Access
Error 502: {extract.model_id.max_length}

**Comment by stevemar** [comment]
@rhagarty can you propose messages that you would like to see?

**Comment by rhagarty** [comment]
I would prefer keeping the existing messages. This PR, however, does handle an additional error case - server not responding. I think that part of the PR needs to be merged.

@markstur - thoughts?

**Comment by rameshpoomalai** [comment]
@rhagarty ,I have updated the code to Match with the previous message format. Last time I was printing only the error message and removed the package details, because that expose the API details. Now I have matched the format with previous version and delivered.

**Comment by rhagarty** [comment]
@rameshpoomalai - the error for invalid credentials looks good, but the one for invalid model ID still needs some work. It shows as "Error:com.ibm.watson.developer_cloud.service.exception.InternalServerErrorException: : {extract.model_id.max_length}

Can you please change this to "Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found"?

**Comment by rameshpoomalai** [comment]
Hi @rhagarty ,
 It looks you are testing with the model with more than the accepted length. if i copy and past same id one more time i could reproduce this error, otherwise i am getting following error.
"Error:com.ibm.watson.developer_cloud.service.exception.NotFoundException: model not found ",

if required i can add validation for model length and handle it, let me know if required?

Thanks
Ramesh

**Comment by rhagarty** [comment]
@rameshpoomalai - yes, you are correct - I had too many chars in my ID. A bad ID gives me the correct error. Thanks for your patience on this.

**Review by rhagarty — APPROVED** [review]

**Pull request merged**